executable/executable.go

```diff
@@ -256,7 +256,7 @@ func (e *Executable) Wait() (ExecutableResult, error) {
 
 // Kill terminates the program
 func (e *Executable) Kill() error {
-	if e.cmd.Process == nil {
+	if !e.isRunning() || e.cmd.Process == nil {
```

Comment:
@ryan-gang under what circumstances would `e.cmd.Process` be nil?

Reply:
Don't think that is possible. 
If e.cmd is nil e.cmd.Process wouldn't be set. Start would fail. 
e.cmd being set properly, would lead to e.cmd.Process to being set on Start().

https://pkg.go.dev/os/exec#Cmd.Start

Reply:
Yeah, seems like that's a case we shouldn't hit then! Feel free to remove, if we hit it we'll get to know via alerts and can then re-think